packages/core/src/mesh/PrimitiveMesh.ts

```diff
@@ -336,6 +456,352 @@ export class PrimitiveMesh {
     PrimitiveMesh._initialize(sphereMesh, vertices, indices, noLongerAccessible, isRestoreMode, restoreVertexBuffer);
   }
 
+  /**
+   * @internal
+   */
+  static _subdivCatmullClark(step: number): {
+    cells: Float32Array;
+    positions: Float32Array;
+  } {
+    const { _tempVec31: tempVec1, _tempVec32: tempVec2, _tempVec33: tempVec3 } = PrimitiveMesh;
+
+    const points: Array<IPoint> = [];
+    const edges: Map<number, IEdge> = new Map();
+    const faces: Array<IFace> = [];
+
```

Comment:
Uniform use of type inference：

```suggestion
     const points = new Array<IPoint>();
    const edges = new Map<number, IEdge>();
    const faces = new Array<IFace>();
```

Reply:
done